airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java

```diff
@@ -48,8 +49,16 @@ public StreamId buildStreamId(final String namespace, final String name, final S
 
   @Override
   public ColumnId buildColumnId(final String name) {
+    return buildColumnId(name, "");
+  }
+
+  @Override
+  public ColumnId buildColumnId(final String name, final String suffix) {
     // No escaping needed, as far as I can tell. We quote all our identifier names.
-    return new ColumnId(escapeSqlIdentifier(name).toUpperCase(), name, name.toUpperCase());
+    final String nameWithSuffix = name + suffix;
+    return new ColumnId(prefixReservedKeyword(escapeSqlIdentifier(name).toUpperCase()) + suffix,
```

Comment:
I think this will be a breaking change - some column names that were previously working  (e.g. `WHERE`) will now be prefixed with `_`. I don't think we need an explicit migration, since we'll auto soft reset, but this should be a major version bump, probably. Unless we think the impact is really small? :shrug:

Or... we can track down which words are only reserved as column names? and define a `RESERVED_COLUMN_NAME_LIST` for them specifically. Then it would be a patch bump (since we were crashing on those column names, so there was no previous behavior)

Reply:
Thanks for calling this out. Going with the second approach to not introduce another breaking change unnecessarily. The list of reserved column names is also relatively small.